### PR TITLE
Upgrade to Android Studio 3.0

### PR DESCRIPTION
**Pull request by manjunathc23**
__BEFORE__

```
BUILD SUCCESSFUL
Total time: 3 mins 16.706 secs
BUILD SUCCESSFUL
Total time: 20.178 secs
```

__AFTER__

```
BUILD SUCCESSFUL in 2m 15s
BUILD SUCCESSFUL in 15s
```

Android Studio 3.0 is amazing!

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -37,31 +51,35 @@ android {
...
 
+    // Specifies a flavor dimension.
+    flavorDimensions "release"
+
     productFlavors {
 
         free {
             applicationIdSuffix ".free"
             versionCode 3
             versionName "1.0.3.1"
```
Comment (cdflynn) [inline app/build.gradle:71]: I see this is `1.0.3.1` and paid is `1.0.3.2`.  Intentional?

File: app/build.gradle [hunk 2]
```diff
@@ -110,97 +150,107 @@ android {
             events 'passed', 'skipped', 'failed', 'standardOut', 'standardError'
             exceptionFormat 'full'
         }
+        //Print the test result summary
+        afterSuite { desc, result ->
+            if (!desc.parent) { // will match the outermost suite
+                println "Myntra Test Results: ${result.resultType} (${result.testCount} tests, ${result.successfulTestCount} successes, ${result.failedTestCount} failures, ${result.skippedTestCount} skipped)"
```
Comment (cdflynn) [inline app/build.gradle]: `Myntra` or `Zeta`?

File: app/build.gradle [hunk 3]
```diff
@@ -110,97 +150,107 @@ android {
...
-    compile libraries.butterKnife
+    implementation libraries.supportV13
+    implementation libraries.supportDesign
+    implementation libraries.supportCardView
+    implementation libraries.supportMultiDex
+    implementation libraries.supportAppCompat
+    implementation libraries.supportRecyclerView
+
+    implementation libraries.eventBus
+    implementation libraries.butterKnife
```
Comment (cdflynn) [inline app/build.gradle:181]: Neat.  Am I reading this right?:
>New api, implementation, compileOnly and runtimeOnly configurations allow libraries to not always leak their dependencies to their consumers.
https://developer.android.com/studio/preview/features/new-android-plugin.html

Just curious how this is different from `compile`, if you happen to know.

File: dependencies.gradle [hunk 4]
```diff
@@ -1,178 +1,238 @@
...
+        ormliteVersion                        : "4.48",
+        prettyTimeVersion                     : "3.2.7.Final",
+
+        autoValueVersion                      : "1.4.1",
+        autoValueJakeWhartonVersion           : "1.4",
+        autoValueParcelVersion                : "0.2.5",
+
+        daggerVersion                         : "2.10",
+        retrofitVersion                       : "2.1.0",
+        rxJavaVersion                         : "1.2.4",
```
Comment (cdflynn) [inline dependencies.gradle:30]: We should make the jump to 2.x at some point.

**Review by cdflynn — APPROVED** [review]
Thanks @manjunathc23!

File: dependencies.gradle [hunk 5]
```diff
@@ -1,178 +1,238 @@
...
-        gradlePluginVersion               : "2.3.1",
-        retroLambdaPluginVersion          : "3.6.1",
-        retrolombokPluginVersion          : "0.2.3.a2",
-        frodoPluginVersion                : "0.8.3"
+        compileSdk                            : 25,
+        minSdk                                : 16,
+        targetSdk                             : 24,
+        buildTools                            : "25.0.3",
+
+        supportLibVersion                     : "25.3.1",
```
Comment (manijshrestha) [inline dependencies.gradle]: even bump it up to 25.4.0?

File: dependencies.gradle [hunk 6]
```diff
@@ -1,178 +1,238 @@
...
+        powerMockitoVersion                   : '1.6.2',
 ]
 
 ext.gradlePlugins = [
+        fabricPlugin      : "io.fabric.tools:gradle:$versions.fabricPluginVersion",
         androidToolsPlugin: "com.android.tools.build:gradle:$versions.gradlePluginVersion",
-        retroLambdaPlugin : "me.tatarka:gradle-retrolambda:$versions.retroLambdaPluginVersion",
         frodoPlugin       : "com.fernandocejas.frodo:frodo-plugin:$versions.frodoPluginVersion",
+        gmsServicesPlugin : "com.google.gms:google-services:$versions.gmsServicesPluginVersion",
+        aptPlugin         : "com.neenbedankt.gradle.plugins:android-apt:$versions.aptPluginVersion",
```
Comment (manijshrestha) [inline dependencies.gradle:86]: any reason we are bringing this back? We are using `annotationProcessor`

**Review by manijshrestha — APPROVED** [review]

File: app/build.gradle (hunk 1 above)
Comment (manjunathc23) [reply app/build.gradle:71]: We can choose any version - I just left it "as is"

File: app/build.gradle (hunk 2 above)
Comment (manjunathc23) [reply app/build.gradle]: Oops, Copy paste error!

File: app/build.gradle (hunk 3 above)
Comment (manjunathc23) [reply app/build.gradle:181]: Here is the session regarding gradle improvements.
https://www.youtube.com/watch?v=7ll-rkLCtyk&t=684s

This year's Google IO tips - `implementation` don't recompile its dependencies. Hence its bit faster in build time.

File: dependencies.gradle (hunk 5 above)
Comment (manjunathc23) [reply dependencies.gradle]: Will do! - We shall go support lib 26 in another PR.

File: dependencies.gradle (hunk 4 above)
Comment (manjunathc23) [reply dependencies.gradle:30]: :) Its a breaking change so refrained from doing so.

File: dependencies.gradle (hunk 6 above)
Comment (manjunathc23) [reply dependencies.gradle:86]: Yeah! May be copy paste error again. Anyways now that there is appt2 we should check that out.

**Review by manjunathc23 — COMMENTED** [review]
Changes done @manijshrestha , @cdflynn 

Thanks for the review.

**Comment by manjunathc23** [comment]
Please review the latest changes! @manijshrestha, @cdflynn

File: app/build.gradle (hunk 3 above)
Comment (cdflynn) [reply app/build.gradle:181]: Nice, thanks for the link!

**Comment by cdflynn** [comment]
👍  @manjunathc23

**Comment by manijshrestha** [comment]
:shipit:

**Comment by manjunathc23** [comment]
This was rebased and merged. Thank you @cdflynn , @manijshrestha

**Pull request closed without merging**